API/Controllers/PortfolioController.cs

```diff
@@ -0,0 +1,498 @@
+                {
+                    Title = "Failed getting the portfolio.",
+                    Detail = "The user id is less then zero and therefore cannot exist in the database.",
+                    Instance = "F7669BF7-CE70-47C0-8D29-37516BA887C0",
+                };
+                return BadRequest(problem);
+            }
+
+            Portfolio portfolio = await portfolioService.FindAsync(portfolioId);
+            if(portfolio == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "the portfolio doesn't exist.",
+                    Detail = "The portfolio id could not be found in the database.",
+                    Instance = "91A15229-CD31-4D6E-920E-9DD6889333F1"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<Portfolio, PortfolioResourceResult>(portfolio));
+        }
+
+        /// <summary>
+        /// This method is responsible for creating a portfolio.
+        /// </summary>
+        /// <returns>This method returns the portfolio resource result.</returns>
+        /// <response code="200">This endpoint returns the created portfolio.</response>
+        /// <response code="400">The 400 Bad Request status code is returned when the portfolio
+        /// resource is not specified or failed to save portfolio to the database.</response>
+        [HttpPost]
+        [Authorize]
+        [ProducesResponseType(typeof(PortfolioResourceResult), (int) HttpStatusCode.Created)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> CreatePortfolioAsync(PortfolioResource portfolioResource)
+        {
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+
+            if(await userService.FindAsync(user.Id) == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting the user account.",
+                    Detail = "The database does not contain a user with this user id.",
+                    Instance = "C199E422-D4BA-49DC-813F-B529959C6AD2"
+                };
+                return NotFound(problem);
+            }
+
+            if(portfolioResource == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to create a new portfolio.",
+                    Detail = "The specified portfolio resource was null.",
+                    Instance = "0A2C218D-CA78-4384-A6A7-0C8C7C01D56B"
+                };
+                return BadRequest(problem);
+            }
+
+            Portfolio portfolio = mapper.Map<PortfolioResource, Portfolio>(portfolioResource);
+
+            try
+            {
+                portfolio.User = user;
+                portfolioService.Add(portfolio);
```

Comment:
I think you can use the overloaded async version here.
```suggestion
                await portfolioService.AddAsync(portfolio);
```